sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -163,22 +165,29 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
 
         this.pollInterval = pollInterval;
         this.pollOperation = pollOperation;
-        this.pollResponse = new PollResponse<>(OperationStatus.NOT_STARTED, null);
+
+        // When the first item is emitted, we set the poll response to it. So the first invocation of pollOperation can
+        // leverage this value.
+        final Mono<T> onActivation = activationOperation == null
+            ? Mono.empty()
+            : activationOperation.get().map(response -> {
+                pollResponse.set(new PollResponse<>(OperationStatus.NOT_STARTED, response));
```

Comment:
`pollResponse` is already initialized with `NOT_STARTED` when the field is declared. Do we need to reinitialize it here?